### ISSUE TITLE
New assets are not appearing 

### ISSUE THREAD

**Issue by tiapatel** [pnp/custom-learning-office-365#65]
Hello,

We have been creating custom playlists with custom assets without an issue. However, today we have found that not all the assets are showing in newly created playlists, neither for end users nor in the Custom Learning Administration area.

In the screenshot below, I have added 7 custom assets to a custom playlist, however only 1 is appearing. I can move the asset up and down, but cannot see the items that I moving it relative to.

![MicrosoftTeams-image](https://user-images.githubusercontent.com/50412225/58243885-71646400-7d49-11e9-9409-1fec9accb139.png)

Do you know what is causing this?

**Comment by FormerlyJelly** [comment]
Hello!

This can be caused by having the web part deployed multiple times as duplicates, or running an older version of the software alongside a newer version. Could you try updating the solution where you are having issues, and also deploy a fresh install to a new communications site to test separately? 

More information about updating the solution can be found here:
https://github.com/pnp/custom-learning-office-365/blob/master/README.md

Kind regards,

Ellis

**Comment by tiapatel** [comment]
Hello! Thanks for getting back to me.

We only have one deployment of the web part. We have updated the software to the latest version available. We are still having the same issue. 

Is there anything else that can be done? The site is currently unusable for us.

**Comment by juliemturner** [comment]
Please confirm the version of the web part you're using?
You can validate it by following the instructions in the [Updating the solution](https://github.com/pnp/custom-learning-office-365/blob/master/README.md#updating-the-solution) section of the README.md

**Comment by tiapatel** [comment]
Version: 2.0.14.0

**Comment by juliemturner** [comment]
I've tried to replicate this issue and am unable to do so. The only thing I can think of is that potentially somehow the entries in the CustomAsset's list were changed and/or removed, or the custom playlist itself was changed manually in the JSONData field of the CustomPlaylists list.  I lean towards the former as it seems as if the system thinks the playlist has multiple assets or you wouldn't see the ability to move the playlist up/down in the order.  You can use the search feature to add custom assets to an existing playlist, can you see if the assets you can't find are showing up in that list?  If you feel comfortable looking at and reading JSON you could also find the custom playlist in the CustomPlaylists list and validate the id's that are specified in the "Assets" array of the JSON string stored in the JSONData field exist in the CustomAssets list.
![CLO365Issue65](https://user-images.githubusercontent.com/7570936/58506533-9566e180-815d-11e9-8f08-de8be336e1ad.png)

**Comment by tiapatel** [comment]
Hello!

Thank you for getting back to me. It appears that the Custom Playlists are including the Custom Asset IDs correctly, but the JSONData for the Custom Assets is not being generated correctly. Please see below. 

![MicrosoftTeams-image (2)](https://user-images.githubusercontent.com/50412225/58626022-a59ccf00-82cb-11e9-9a39-aea4ad5a3a45.png)
![MicrosoftTeams-image (1)](https://user-images.githubusercontent.com/50412225/58626023-a6356580-82cb-11e9-865a-a8e5058ffefd.png)


What do you recommend that we do?

**Comment by juliemturner** [comment]
That's actually not a bug that's exactly right, the fact that the ID in the JSON string is 0 for the assets is fine.  What you're not showing me is that all 5 items are in your assets list, but I assume you've checked that (in other words your screen shot only shows 116 and 118)... although that is more confusing because you said you had 7 assets in the playlist but the JSON shows only 5.  I guess the question is are you sure this is the right playlist item? Are you sure that all 5 assets with those id's exists?  

Assuming yes, I guess I would suggest you just delete this playlist item from the custom playlists list and recreate it through the UI. Add the assets back to the playlist by using search instead of add new since they already exist. If you can provide some details about how to reproduce your issue please update the post or if it's been closed open a new one.

**Comment by tiapatel** [comment]
I have noticed that the assets that any Custom Asset after id 99, have the JSON string 0, and it is those that are not appearing in the playlists or causing that issue. Is the 0 after 99 Assets expected?

I have tried to create new Custom Playlists with new Custom Assets and the issue persists.


<img width="454" alt="CustomAssets" src="https://user-images.githubusercontent.com/50412225/58628620-1cd56180-82d2-11e9-991e-1cc183a9d3a3.png">
